Lib/ufo2ft/util.py

```diff
@@ -484,6 +484,33 @@ def _loadPluginFromString(spec, moduleName, isValidFunc):
     return klass(**options)
 
 
+def overrideSourceNamesFromDesignSpace(font, designspace_entry, inplace=False):
+    """Allow overriding the family and style name in a font
+    if its designspace entry specifies custom values"""
+
+    if not inplace:
+        font_class = font.__class__
+        if hasattr(font_class, "getDataForSerialization"):
+            # work around defcon.Font deepcopy
+            data = font.getDataForSerialization()
+            font = font_class()
+            font.setDataFromSerialization(data)
+        else:
+            font = deepcopy(font)
```

Comment:
you're doing the copy (which is relatively expensive operation) before knowing if the copy is needed, that is before checking if the designspace overrides familyName or styleName.
I suggest you first check that and only copy if either is not None (and inplace is False of course)

Reply:
Fixed.